src/configuration/index.js

```diff
@@ -0,0 +1,205 @@
+        }
+
+        try {
+            const optionsObj = JSON.parse(configurationFileContent);
+
+            this._options = Configuration._fromObj(optionsObj);
+        }
+        catch (e) {
+            console.log(ERR_CONFIG_FILE_IS_NOT_WELLFORMATTED); // eslint-disable-line no-console
+        }
+
+        await this._prepareOptions();
+    }
+
+    async _prepareOptions () {
+        await this._prepareSslOptions();
+        this._prepareFilterFn();
+        this._ensureArrayOption(OPTION_NAMES.src);
+        this._ensureArrayOption(OPTION_NAMES.browsers);
+        this._ensureArrayOption(OPTION_NAMES.reporter);
+    }
+
+    _ensureArrayOption (name) {
+        const options = this._options[name];
+
+        if (!options)
+            return;
+
+        options.value = castArray(options.value);
+    }
+
+    _prepareFilterFn () {
+        const filterOption = this._ensureOption(OPTION_NAMES.filter, null);
+
+        if (!filterOption.value)
+            return;
+
+        const { test, fixture, testGrep, fixtureGrep, testMeta, fixtureMeta } = filterOption;
+
+        const opts = {
+            testGrep:    optionValueToRegExp('testGrep', testGrep),
+            fixtureGrep: optionValueToRegExp('FixtureGrep', fixtureGrep),
+            test,
+            fixture,
+            testMeta,
+            fixtureMeta
+        };
+
+        filterOption.value = createFilterFn(opts);
+    }
+
+    async _prepareSslOptions () {
+        const sslOptions = this._options[OPTION_NAMES.ssl];
+
+        if (!sslOptions)
+            return;
+
+        await Promise.all(Object.entries(sslOptions.value).map(async entry => {
+            const key   = entry[0];
+            const value = entry[1];
+
+            sslOptions.value[key] = await ensureSslOptionValue(key, value);
+        }));
+    }
+
+    _ensureOption (name, value, source) {
+        let option = null;
+
+        if (name in this._options)
+            option = this._options[name];
+        else {
+            option = new Option(name, value, source);
+
+            this._options[name] = option;
+        }
+
+        return option;
+    }
+
+    mergeOptions (options) {
+        const overridenOptions = [];
+
+        Object.entries(options).map(item => {
+            const key    = item[0];
+            const value  = item[1];
+            const option = this._ensureOption(key, value, optionSource.input);
+
+            if (value === void 0)
+                return;
+
+            if (option.value !== value &&
+                option.source === optionSource.configuration)
+                overridenOptions.push(key);
+
+            option.value  = value;
+            option.source = optionSource.input;
+        });
+
+        if (overridenOptions.length)
+            console.log(`${overridenOptions.map(option => `"${option}"`).join(', ')} option${overridenOptions.length > 1 ? 's' : ''} from configuration file will be ignored.`); // eslint-disable-line no-console
```

Comment:
We should keep all strings in constants.

Reply:
We can store it in the `notifications/warning-messages` and then use `renderWarning` to render a string for output.

Reply:
Ok

Reply:
@AndreyBelym, I don't think the outputting this message as a warning is a good idea. It's just a log message.